HW8/src/ArrayValueCalculator.java

```diff
@@ -0,0 +1,63 @@
+import exceptions.ArrayDataException;
+import exceptions.ArraySizeException;
+
+public class ArrayValueCalculator {
+
+    private static final StringBuilder incorrectData = new StringBuilder();
+
+    public static void main(String[] args) {
+        String[][] arrayForCalc = new String[][] {
+            {"21", "()*)", "33", "41"},
+            {"35", "**", "27", "18"},
+            {"19", "10", "11", "*913"}
+            /*, {"9", "10", "11", "12"}*/
+        };
+        int dimension = 4;
+
+        System.out.println("\nResult of doCalc():");
+        System.out.println(doCalc(arrayForCalc, dimension));
+    }
+
+    private static int doCalc(String[][] arrayForCalc, int dimension)
+        throws ArraySizeException, ArrayDataException {
+
```

Comment:
First, there should be an array size check that could throw an exception if the array has the wrong dimensions. Something like it:
```
if(isArrayDimmentionCorrect(arrayForCalc){
  if(dimension is not correct== true) throw new ArraySizeException(" meaningful 
message");
}
```